docs/debugging-javascript.md

```diff
@@ -0,0 +1,382 @@
+    {
+        "name": "Debug Windows Hermes - Experimental",
+        "cwd": "${workspaceFolder}",
+        "type": "reactnativedirect",
+        "request": "launch",
+        "platform": "windows",
+        "isDynamic": true
+    }
+    ```
+    3. Ensure that Metro and your native app are **not** already running
+    4. Ensure the new config is selected at the top of the *Run and Debug* sidebar
+    5. Click on the ▶️ button or press `F5` in VS Code
+    > Watch VS Code's *Debug Console* output for a successful debugger connection. This can be very slow, but VS Code should automatically launch Metro, establish the debugger connection, then launch your native app to download the bundle.
+6. **Option B:** Attach VS Code to Metro that's already running
+    1. Click on the ⚙️ to the right of *Attach to Hermes application - Experimental*
+    2. A new configuration should appear in your `launch.json` file. Rename it if you want but it should look like:
+    ```json
+    {
+        "name": "Attach to Hermes application - Experimental",
+        "cwd": "${workspaceFolder}",
+        "type": "reactnativedirect",
+        "request": "attach",
+        "isDynamic": true
+    }
+    ```
+    3. Ensure that Metro is already running (i.e. `npx react-native start`)
+    4. Ensure the new config is selected at the top of the *Run and Debug* sidebar
+    5. Click on the ▶️ button or press `F5` in VS Code
+    6. Confirm the address of the Metro server (default: `localhost`)
+    7. Confirm the port of the Metro server (default: `8081`)
+    > Watch VS Code's *Debug Console* output for a successful debugger connection.
+
+    8. Manually launch your native app (if it wasn't already running)
+
+Once everything is running, you should be able to debug your RNW application. To set breakpoints, simply do so in your app's JS source files directly in VS Code.
+
+> Remember to save `launch.json` so you can reuse this debugging configuration in the future.
+
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+## Web vs. Direct Debugging
+
+As mentioned above, RNW supports both Web and Direct Debugging, and it's worth knowing when to use each.
+
+Web Debugging is the older, more well-established of the two debugging solutions. It relies on the powerful in-browser debugging tools already familiar to web developers. This reinforces RN's purpose of enabling web developers to build native apps with web technologies.
+
+However, Web Debugging is not without its limitations. The primary problem is that JS can behave slightly differently when running in different JS engines. As such, by developing (and testing) your app while using a browser to run the JS, you might miss bugs or other issues that only manifest when running your app normally with the embedded JS engine.
+
+This puts extra onus on developers to test all of their app's functionality in their release build, or otherwise risks situations where customers find bugs in production that aren't reproducible when using Web Debugging.

```

Comment:
>onus [](http://example.com/codeflow?start=16&length=4)

Could we use an alternative word? It seems to be a quite rare word which I personally see the first time. The docs should be readable not only to native English speakers.